src/gs-shell.c

```diff
@@ -1963,6 +1963,9 @@ gs_shell_setup (GsShell *shell, GsPluginLoader *plugin_loader, GCancellable *can
 
 	/* coldplug */
 	gs_shell_rescan_events (shell);
+
+	/* show loading page, which triggers the initial refresh */
+	gs_shell_change_mode (shell, GS_SHELL_MODE_LOADING, NULL, TRUE);
```

Comment:
Isn't this commit equivalent to what was being done (and hence not necessary)? AFAICS only the application calls the `gs_shell_setup`, so these changes seem to only call the mentioned method before the `_add_window`.

Reply:
Correct, it doesn't change the behaviour, although if the code I deleted in a77b763 hadn't been so broken, this would never have worked - which for a while I thought was the bug. Per above it feels more consistent to me that the shell would advance itself through the state machine rather than splitting the state advances between two classes.